### PR TITLE
Mixture of non-prefixed and prefixed (B-, I-)

### PR DESCRIPTION
**Pull request by KoichiYasuoka**
# What does this PR do?

Reconsider of #13493 for mixture case, where B-TAG and I-TAG are used for multi-tokens and TAG is used for a single token.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@Narsil

### CODE REVIEW

**Comment by Narsil** [comment]
This seems a bit overboard. I don't think mixed tags is desirable. (it's very uneasy to know what a model will do.
The previous fix was about being backward compatible, this is a proposed breaking change I think.

However I am not necessarily having all information.

@LysandreJik do you have an opinion ?

**Comment by LysandreJik** [comment]
Pinging @stefan-it as he has extensive experience with NER, maybe you can enlighten us? :)

**Comment by stefan-it** [comment]
Yeah, I think this is also related to this issue:

https://github.com/huggingface/transformers/issues/4262

tldr: some of the NER datasets use IOB1 as labeling scheme (namely CoNLL-2003 English and German).

**Comment by Narsil** [comment]
Hi @stefan-it , but reading the conversation it just confirms that using I- only is the way to go, and that we cannot have same but separated entity splits with this scheme. 

It seems that the proposed PR wouldn't actually solve the problem either because B- are never outputted, right ?
